src/plugins/discover/public/application/components/table/table.tsx

```diff
@@ -117,31 +119,34 @@ export const DocViewerTable = ({
       const nameB = !mappingB || !mappingB.displayName ? fieldB : mappingB.displayName;
       return nameA.localeCompare(nameB);
     })
-    .map((fieldName) => {
-      const fieldMapping = mapping(fieldName);
-      const fieldType = isNestedFieldParent(fieldName, indexPattern)
-        ? 'nested'
-        : fieldMapping?.type;
+    .map((field) => {
+      const fieldMapping = mapping(field);
+      const fieldKey = (field ? field : fieldMapping?.displayName) as string;
```

Comment:
I would not recommend to use the `displayName` as `fieldKey`, since it can be customized by the user in the index pattern. better doing it his way

```suggestion
      const fieldKey = field;
      const fieldName = fieldMapping?.displayName ?? field;
```
fieldName ist just used for what's being displayed (also naming it `const displayName` would be an option, since then it should be clear what it's doing)

of course this would mean adapting all the follow up code, but it would be an improvement IMO. because currently it's: confusing

Reply:
Yes, we can keep just `field` and `displayField`, thank you.